File: scripts/convert-photos.js
const fs = require("fs");
const path = require("path");
const sharp = require("sharp");

const PHOTOS_DIR = path.join(__dirname, "../public/photos");
const manifestPath = path.join(PHOTOS_DIR, "manifest.json");

async function convert() {
  const manifest = JSON.parse(fs.readFileSync(manifestPath, "utf-8"));
  const newManifest = [];

  for (const file of manifest) {
    const inputPath = path.join(PHOTOS_DIR, file);
    if (!fs.existsSync(inputPath)) {
      console.log(`Skip (not found): ${file}`);
      continue;
    }
    const baseName = path.basename(file, path.extname(file));
    const outputName = `${baseName}.webp`;
    const outputPath = path.join(PHOTOS_DIR, outputName);

    await sharp(inputPath)
      .resize(800, null, { withoutEnlargement: true })
      .webp({ quality: 80 })
      .toFile(outputPath);

    newManifest.push(outputName);
    console.log(`Converted: ${file} -> ${outputName}`);
  }

  fs.writeFileSync(manifestPath, JSON.stringify(newManifest, null, 2));
  console.log("Done! manifest.json updated.");
}

convert().catch(console.error);

File: convert_photos.py
from PIL import Image
import os, json, glob

DIR = "public/photos"

files = sorted(
    glob.glob(os.path.join(DIR, "*.jpg")) +
    glob.glob(os.path.join(DIR, "*.JPG")) +
    glob.glob(os.path.join(DIR, "*.jpeg")) +
    glob.glob(os.path.join(DIR, "*.JPEG")) +
    glob.glob(os.path.join(DIR, "*.png")) +
    glob.glob(os.path.join(DIR, "*.PNG"))
)

new_manifest = []
for path in files:
    fname = os.path.basename(path)
    base = os.path.splitext(fname)[0]
    out_name = base + ".webp"
    out_path = os.path.join(DIR, out_name)
    
    img = Image.open(path)
    if img.mode in ("RGBA", "P"):
        img = img.convert("RGB")
    
    if img.width > 800:
        ratio = 800 / img.width
        new_h = int(img.height * ratio)
        img = img.resize((800, new_h), Image.LANCZOS)
    
    img.save(out_path, "webp", quality=80, method=6)
    new_manifest.append(out_name)
    print(f"Converted: {fname} -> {out_name} ({img.width}x{img.height})")

manifest_path = os.path.join(DIR, "manifest.json")
with open(manifest_path, "w", encoding="utf-8") as f:
    json.dump(new_manifest, f, indent=2)

print(f"\nDone! {len(new_manifest)} images converted. manifest.json updated.")